tilequeue/utils.py

```diff
@@ -23,3 +26,51 @@ def grouper(iterable, n):
         if not chunk:
             return
         yield chunk
+
+def parse_log_file(log_file):
+    ip_pattern = '(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'
+    # didn't match againts explicit date pattern, in case it changes
+    date_pattern = '\[([\d\w\s\/:]+)\]'
+    tile_id_pattern = '\/([\w]+)\/([\d]+)\/([\d]+)\/([\d]+)\.([\d\w]*)'
+
+    log_pattern = '%s - - %s "([\w]+) %s.*' % (ip_pattern, date_pattern, tile_id_pattern)
+
+    tile_log_records = []
+    for log_string in log_file:
+        match = re.search(log_pattern, log_string)
+        if match and len(match.groups()) == 8:
+            tile_log_records.append((match.group(1), 
+                                      datetime.strptime(match.group(2), '%d/%B/%Y %H:%M:%S'), 
+                                      coord_marshall_int(create_coord(match.group(6), match.group(7), match.group(5)))))
+
+    return tile_log_records
+
+def mimic_prune_tiles_of_interest_sql_structure(cursor):
+    cursor.execute('''CREATE TABLE IF NOT EXISTS tile_traffic_v4 (
+            id bigserial primary key,
+            date timestamp(6) not null,
+            z integer not null,
+            x integer not null,
+            y integer not null,
+            tilesize integer not null,
+            service varchar(32),
+            host inet not null
+        )''')
+
+def postgres_add_compat_date_utils(cursor):
```

Comment:
I thought we had found something that worked on both platforms in a previous discussion, but looking back I see that's not the case.

It looks like both [RedShift](http://docs.aws.amazon.com/redshift/latest/dg/r_interval_literals.html) and [PostgreSQL](https://www.postgresql.org/docs/9.1/static/datatype-datetime.html#DATATYPE-INTERVAL-INPUT) support interval literals:

#### On PostgreSQL

```sql
iandees=# select current_timestamp - interval '30 days' as dateplus;
           dateplus
-------------------------------
 2017-05-06 11:12:11.184408-04
(1 row)
```
#### On RedShift

```sql
analytics=# select current_timestamp - interval '30 days' as dateplus;
           dateplus
-------------------------------
 2017-05-06 15:12:20.733685+00
(1 row)
```

Can you change to using this interval literal notation in both cases and remove the code to add a wrapping function?